### PR TITLE
test(shared/helpers): add unit tests for shared helpers

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Improving test coverage; finding any edge cases.

_Powered by Claude under my oversight._

## Changes proposed in this pull request

Add tests for `src/shared/helpers/*`. To avoid making the test file too large, map directory structure for new tests (keeping older tests in helpers.test.ts)

### CODE REVIEW

File: src/shared/__tests__/helpers-i18n.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  return {
+    i18n: {
+      getMessage: vi.fn<Browser['i18n']['getMessage']>(() => 'translated'),
+    },
+  };
+}
+
+// the fake above only implements the slice of Browser['i18n'] tFactory needs
+type FakeI18nBrowser = ReturnType<typeof makeI18nBrowser>;
+const asI18nBrowser = (browser: FakeI18nBrowser) =>
```
Comment (MichaelKohler) [inline src/shared/__tests__/helpers-i18n.test.ts]: Similar comment to the other PR, there already is a `makeBrowser` function that could be extended for this.

**Review by MichaelKohler — APPROVED** [review]
